server/api.js

```diff
@@ -0,0 +1,409 @@
+  if (button == null || button.client.id !== req.params.clientId) {
+    res.status(404).send({ status: 'error', message: 'Not Found' })
+
+    return
+  }
+
+  // attempt to update the button
+  const updatedButton = await db.updateButton(
+    req.body.clientId,
+    req.body.displayName,
+    req.body.phoneNumber,
+    req.body.buttonSerialNumber,
+    req.body.isDisplayed,
+    req.body.isSendingAlerts,
+    req.body.isSendingVitals,
+    req.params.buttonId,
+  )
+
+  // something bad happened and the button wasn't updated; blame it on the request
+  if (updatedButton == null) {
+    res.status(400).send({ status: 'error', message: 'Bad Request' })
+
+    return
+  }
+
+  res.status(200).send({ status: 'success', data: updatedButton })
+}
+
+// NOTE: clientId is submitted in the param and body of the request.
+// This is to let a gateway be moved from one client to another; think of the param clientId as 'from' and the body clientId as 'to'.
+const validateUpdateClientGateway = [
+  Validator.param(['clientId', 'gatewayId']).notEmpty(),
+  Validator.body(['clientId', 'displayName']).trim().isString().notEmpty(),
+  Validator.body(['isDisplayed', 'isSendingVitals']).trim().isBoolean(),
+]
+
+async function handleUpdateClientGateway(req, res) {
+  const gateway = await db.getGatewayWithId(req.params.gatewayId)
+
+  // check that this gateway exists and is owned by the specified client
+  // NOTE: if clientId is invalid, then the query will fail and return null
+  if (gateway == null || gateway.client.id !== req.params.clientId) {
+    res.status(404).send({ status: 'error', message: 'Not Found' })
+
+    return
+  }
+
+  // attempt to update the gateway
+  const updatedGateway = await db.updateGateway(
+    req.body.clientId,
+    req.body.displayName,
+    req.body.isDisplayed,
+    req.body.isSendingVitals,
+    req.params.gatewayId,
+  )
+
+  // something bad happened and the gateway wasn't updated; blame it on the request
+  if (updatedGateway == null) {
+    res.status(400).send({ status: 'error', message: 'Bad Request' })
+
+    return
+  }
+
+  res.status(200).send({ status: 'success', data: updatedGateway })
+}
+
+module.exports = {
+  handleCreateClient,
+  handleCreateClientButton,
+  handleCreateClientGateway,
+  handleGetClient,
+  handleGetClients,
+  handleGetClientButton,
+  handleGetClientButtons,
+  handleGetClientButtonSessions,
+  handleGetClientGateway,
+  handleGetClientGateways,
+  handleGetClientVitals,
+  handleUpdateClient,
+  handleUpdateClientButton,
+  handleUpdateClientGateway,
+  validateCreateClient,
+  validateCreateClientButton,
+  validateCreateClientGateway,
+  validateGetClient,
+  validateGetClientButton,
+  validateGetClientButtons,
+  validateGetClientButtonSessions,
+  validateGetClientGateway,
+  validateGetClientGateways,
+  validateGetClientVitals,
+  validateUpdateClient,
+  validateUpdateClientButton,
+  validateUpdateClientGateway,
+  authorize,
+}
```

Comment:
new route for client sessions, which displays recent sessions (maybe last 20? 40?) for a given client